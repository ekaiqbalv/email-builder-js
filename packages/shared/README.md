# @usewaypoint/shared

Shared utilities for the email builder.
