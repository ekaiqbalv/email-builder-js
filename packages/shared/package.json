{
  "name": "@usewaypoint/shared",
  "version": "0.0.1",
  "description": "Shared utilities for the email builder",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "require": "./dist/index.js",
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.ts"
    }
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "npx tsup src/index.ts --format cjs,esm --dts"
  },
  "author": "Eka Iqbal Virgiawan",
  "license": "MIT",
  "devDependencies": {
    "@types/jest": "^29.5.14",
    "jest": "^29.7.0",
    "ts-jest": "^29.2.5"
  }
}
