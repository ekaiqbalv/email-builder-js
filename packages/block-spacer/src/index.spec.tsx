import React from 'react';

import { matchers } from '@emotion/jest';
import { render } from '@testing-library/react';
import { BREAKPOINTS } from '@usewaypoint/shared';

import { Spacer } from '.';

expect.extend(matchers);

describe('Spacer', () => {
  it('renders with default values', () => {
    expect(render(<Spacer />).asFragment()).toMatchSnapshot();
  });

  it('renders with props', () => {
    expect(render(<Spacer props={{ height: 10 }} />).asFragment()).toMatchSnapshot();
  });

  it('applies responsive heights correctly', () => {
    const { container } = render(<Spacer props={{ height: 20 }} />);
    expect(container.firstChild).toHaveStyleRule('height', '16px');
    expect(container.firstChild).toHaveStyleRule('height', '17px!important', {
      media: `(min-width: ${BREAKPOINTS.sm}px)`,
    });
    expect(container.firstChild).toHaveStyleRule('height', '19px!important', {
      media: `(min-width: ${BREAKPOINTS.md}px)`,
    });
    expect(container.firstChild).toHaveStyleRule('height', '20px!important', {
      media: `(min-width: ${BREAKPOINTS.lg}px)`,
    });
  });
});
