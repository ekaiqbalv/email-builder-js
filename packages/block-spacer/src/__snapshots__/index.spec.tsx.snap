// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer renders with default values 1`] = `
<DocumentFragment>
  <div
    class="css-sh1rxn"
  />
</DocumentFragment>
`;

exports[`Spacer renders with props 1`] = `
<DocumentFragment>
  <div
    class="css-av7sfo"
  />
</DocumentFragment>
`;
